compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1665,7 +1665,7 @@ object desugar {
           case t => t
         }
         // This is a deliberate departure from scalac, where StringContext is not rooted (See #4732)
-        Apply(Select(Apply(scalaDot(nme.StringContext), strs), id), elems)
+        Apply(Select(Apply(scalaDot(nme.StringContext).withSpan(tree.span), strs), id), elems)
```

Comment:
We could also try this variant to see if the semanticdb test keep on working without changes.
```suggestion
        Apply(Select(Apply(scalaDot(nme.StringContext), strs), id).withSpan(tree.span), elems)
```

Reply:
Giving that a try, you get https://github.com/lampepfl/dotty/pull/8629/commits/31e946445265bf57aa1d6193deaf470c8222a2e6 

No more truncated string segments, but I have no idea whether that was a problem in the first place or not.

Reply:
Looks better. Maybe @bishabosha can tell us which one is better. If not I would go with this one.

Reply:
I think its better now that the actual method call has a span the length of the interpolator prefix, rather than empty but i guess its not matched exactly to the prefix, but there are other desugaring anomalies like extension methods.
So this is good.